backend/src/main/java/fr/zelytra/session/SessionSocket.java

```diff
@@ -42,41 +44,45 @@ public void onOpen(Session session) {
 
     }
 
+
     @OnMessage
     public void onMessage(String message, Session session, @PathParam("sessionId") String sessionId) throws JsonProcessingException {
 
-        // Cancel the timeout task since we've received the message
-        Future<?> timeoutTask = sessionTimeoutTasks.remove(session.getId());
-        if (timeoutTask != null) {
-            timeoutTask.cancel(true);
-        }
-
         ObjectMapper objectMapper = new ObjectMapper();
         objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
         objectMapper.enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS);
 
-        Player player = objectMapper.readValue(message, Player.class);
-        player.setSocket(session);
-
-        SessionManager manager = SessionManager.getInstance();
-
-        //Check if it's an update player request
-        if (manager.isPlayerInSession(player, player.getSessionId())) {
-
-            Fleet fleet = manager.getFleetFromId(player.getSessionId());
-            assert fleet != null;
-            Player foundedplayer = fleet.getPlayerFromUsername(player.getUsername());
-
-            foundedplayer.setReady(player.isReady());
-            foundedplayer.setStatus(player.getStatus());
+        // Deserialize the incoming message to SocketMessage<?>
+        SocketMessage<?> socketMessage = objectMapper.readValue(message, new TypeReference<>() {
+        });
```

Comment:
Why is that empty ?

Reply:
https://www.baeldung.com/java-record-keyword